pdns/pdns_recursor.cc

```diff
@@ -2348,10 +2365,53 @@ static bool trySendingQueryToWorker(unsigned int target, ThreadMSG* tmsg)
     }
   }
 
+  ++targetInfo.numberOfDistributedQueries;
+
   return true;
 }
 
-// This function is only called by the distributor thread, when pdns-distributes-queries is set
+static unsigned int getWorkerLoad(size_t workerIdx)
+{
+  const auto mt = s_threadInfos[/* skip distributor */ 1 + workerIdx].mt;
+  if (mt != nullptr) {
+    return mt->numProcesses();
```

Comment:
So is this safe? Doesn't this need a lock to protect d_threads in MTasker? Also, I think this would suck, perhaps there is a better way to do this. Also this assumes the load is proportional to the number of processes and not to the number of queries per second - although I am also not sure how we'd measure that.

Reply:
We might have to do a numThreads variable in MTasker and keep it updated.

Reply:
It would be safer to keep a separate variable, updated, yes. I'm pretty sure that's how it's implemented in practice but it's not guaranteed by the standard. I'll update this PR.

Reply:
I agree estimating the load this ways is far from being perfect, but it was the cheapest estimation I could find. We could keep track of the number of queries distributed to each thread, or even the ones actually in the distribution pipe, but it would come at a cost that it's not easy to estimate.